lib/matplotlib/tests/test_collections.py

```diff
@@ -662,6 +689,30 @@ def test_quadmesh_set_array():
     assert np.array_equal(coll.get_array(), np.ones(9))
 
 
+def test_quadmesh_alpha_array():
+    x = np.arange(4)
+    y = np.arange(4)
+    z = np.arange(9).reshape((3, 3))
+    alpha = z / z.max()
+    alpha_flat = alpha.ravel()
```

Comment:
So at somepoint someone will put some masked or NaN into alpha.  What will happen, and what do you recommend?

Reply:
Fail on nan; ignore mask.  I don't think it is worthwhile for us to try to propagate nans and masked arrays from the alpha input.